### ISSUE TITLE
Straub fragmentation function optimization: avoid using Storage `__getitem__`

### ISSUE THREAD

**Issue by slayoo** [atmos-cloud-sim-uj/PySDM#976]
There seem to be a significant overhead in using constructs like:
https://github.com/atmos-cloud-sim-uj/PySDM/blob/ac01b4ea5e91f3045d9ba3acbe61c828ed42d589/PySDM/dynamics/collisions/breakup_fragmentations/straub2010.py#L50
which result in clling `Storage` `__getitem__` millions of times in simulations like the 1D rainshaft one.

A possible workaround would be to implement a method in `Storage` (or `PairwiseStorage` in this case) which would do the copying using some backend code (`njitted` in the case of Numba CPU backend).

**Comment by edejong-caltech** [comment]
@slayoo Agreed we should address this for the sake of speeding up the code -- I don't feel that I understand the backend storage methods well enough to implement an njitted copying function. Is this something that you or @abulenok might be able to help with?

**Comment by slayoo** [comment]
I suggest, we do it after merging #994 (and let's merge it ASAP)

**Comment by slayoo** [comment]
this has been addressed in #1014
